packages/next/src/build/webpack/loaders/next-middleware-loader.ts

```diff
@@ -44,8 +44,9 @@ export default function middlewareLoader(this: any) {
 
         enhanceGlobals()
 
-        var mod = require(${stringifiedPagePath})
-        var handler = mod.middleware || mod.default;
+        import * as _mod from ${stringifiedPagePath}
+        const mod = { ..._mod }
```

Comment:
This shouldn't be needed? It will also deopt any tree-shaking (though this is dynamic built code?), because it won't be able to associate the `mod.middleware` and `mod.default` below as the only accesses.

Reply:
I think we shouldn't have `middleware` exports for edge function loaders as we already have one loader for middleware itself. cc @javivelasco for confirmation